src/gpu/nvidia/cudnn_convolution_impl.hpp

```diff
@@ -581,8 +622,32 @@ struct cudnn_convolution_impl_fwd_t : public cudnn_convolution_impl_base_t {
             CUDA_EXECUTE_FUNC(cuMemcpy, (CUdeviceptr)&host_dst_scale,
                     (CUdeviceptr)dst_scale, sizeof(float));
             float inv_scale = 1.0f / host_dst_scale;
-            CUDNN_EXECUTE_FUNC(
-                    cudnnScaleTensor, handle, descs[io::y], y, &inv_scale);
+            if (data_types[io::y] == CUDNN_DATA_INT8) {
```

Comment:
there is a code section above that handles post operations. For example:
```
                case dnnl_eltwise:
                    if (last_op) {
                        execute_eltwise(handle, output, y);
                    } else {
                        execute_eltwise(handle, output, post_op_scratch);
                    }
                    break;
```
eltwise already writes to `output`, so the dst scaling of `y_fp32_data` after should not work correctly as the eltwise operation itself, because convolution wrote to `y_fp32_data`. What tests do you run and what GPU are you using?

Reply:
I just wanted to let you know that I got your point here. I will double-check it.

I used to run these two batch test files:
```c++
./benchdnn --conv --engine=gpu --batch=test_conv_gpu
./benchdnn --conv --engine=gpu --dt=s8:s8:s8 --dir=FWD_I --skip-impl=ref --attr-scales=src:common:2 --batch=set_conv_all
```
If there is any additional file to test please let me know.
I am using an NVIDIA A100-PCIE-40GB GPU.
Thanks for your feedback.

Reply:
The tests should include post-ops, src and dst scales to validate the code path you modified:
`./benchdnn --conv --engine=gpu --dt=s8:s8:s8 --dir=FWD_I --skip-impl=ref --attr-scales=src:common:2+dst:common:4 --attr-post-ops=eltwise_relu --batch=set_conv_all`
- The src scales should enforce the implementation to compute into f32 accumulator;
- The post-ops will use `tmp_dst`
- The dst scales is what is broken currently.

I would also recommend to play a bit with dst scales value (4 in this example) to make sure that if conversion f32->s8 is before the dst scaling (as it is currently) the test fails due to rounding issues.

Reply:
We took into account the tests you mentioned in this last commit. The reorder is done directly after the scaling. We are not using the `need_reorder` flag. The path that our algorithm is taking is a different one where the reorder process is done after the dst scaling when the function `cudnnOpTensor` is used.

Reply:
Thanks for updating the code, the following test case still fails:
`--conv --engine=gpu --skip-impl=ref --dir=FWD_I --dt=s8:s8:s8 --attr-scales=src:common:2+dst:common:4 --attr-post-ops=sum g6ic24iw11oc24ow9kw3pw0n"1d_gconv_channel-block-4"`
I used the following command to make sure post ops are tested (just a relu postop can be fused into cudnn call, so post ops logic will be skipped, but we want to test it):
` ./build/tests/benchdnn/benchdnn --conv --engine=gpu --dt=s8:s8:s8,s8:s8:f32 --dir=FWD_I --skip-impl=ref --attr-scales=src:common:2+dst:common:4 --attr-post-ops=eltwise_relu,sum --batch=set_conv_all`

Reply:
Thanks for the feedback. I did check the code and applied some changes to avoid the reported issue. I tested locally and the benchdnn tests are passing.

Reply:
Thanks for the update!

There is another class of test cases that fail: no dst scale, but dst is s8:
```./build/tests/benchdnn/benchdnn --conv --engine=gpu --skip-impl=ref --dir=FWD_I --dt=s8:s8:s8 --attr-scales=src:common:2 --attr-post-ops=sum mb1ic512iw121oc512ow122kw6pw3nconv1d:21```

Even dst scale is not set, the post ops computation should happen in f32 and then quantized  but the dst_scale will be equal to `1` (default scale value).

Reply:
Thanks for letting me know this is an issue @igorsafo. I did some fixes to support these cases. In the last commit, I addressed them. If there's something else to check please let me know.